grammars/clojure.cson

```diff
@@ -277,7 +277,8 @@
     'name': 'meta.expression.clojure'
     'patterns': [
       {
-        'begin': '(?<=\\()(ns|def|def-|defn|defn-|defvar|defvar-|defmacro|defmacro-|deftest)\\s+'
+        # ns, declare and everything that starts with def* or namespace/def*
+        'begin': '(?<=\\()(ns|declare|def[\\w\\.\\-\\_\\:\\+\\=\\>\\<\\!\\?\\*\\d]*|[\\w\\.\\-\\_\\:\\+\\=\\>\\<\\!\\?\\*][\\w\\.\\-\\_\\:\\+\\=\\>\\<\\!\\?\\*\\d]*/def[\\w\\.\\-\\_\\:\\+\\=\\>\\<\\!\\?\\*\\d]*)\\s+'
```

Comment:
And yes, colon can’t be the last char in symbol name, but everything else is a valid:

```clojure
user=> (defmacro def+!.?abc8:<> [arg] `(def ~arg))
#'user/def+!.?abc8:<>
user=> (def+!.?abc8:<> x)
#'user/x
user=>
```